engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.content-hosts.controller:ContentHostsBulkSystemPurposeModalController
+ *
+ * @requires $scope
+ * @requires $uibModalInstance
+ * @requires HostBulkAction
+ * @requires Organization
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires hostIds
+ *
+ * @description
+ *   A controller for providing bulk action functionality for setting system purpose values
+ */
+angular.module('Bastion.content-hosts').controller('ContentHostsBulkSystemPurposeModalController',
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.purposeAddonsCount = 0;
+        $scope.purposeAddonsList = [];
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            $scope.purposeAddonsCount += org.system_purposes.addons.length;
+            if ($scope.purposeAddonsCount > 0) {
+                $scope.purposeAddonsList.push(org.system_purposes.addons);
+            }
+        });
+
+        $scope.defaultUsages = ['No change', 'None (Clear)', 'Production', 'Development/Test', 'Disaster Recovery'];
+        $scope.defaultRoles = ['No change', 'None (Clear)', 'Red Hat Enterprise Linux Server', 'Red Hat Enterprise Linux Workstation', 'Red Hat Enterprise Linux Compute Node'];
+        $scope.defaultServiceLevels = ['No change', 'None (Clear)', 'Self-Support', 'Standard', 'Premium'];
+
+        $scope.selectedUsages = $scope.defaultUsages[0];
+        $scope.selectedRoles = $scope.defaultRoles[0];
+        $scope.selectedServiceLevels = $scope.defaultServiceLevels[0];
+
+        $scope.selected = {
+           addons: undefined
+        };
+
+        function actionParams() {
+            var params = hostIds;
+
+            if ($scope.selectedUsages === "None (Clear)") {
+                params['purpose_usage'] = "";
+            } else if ($scope.selectedUsages === "No change") {
+                params['purpose_usage'] = null;
+            } else {
+                params['purpose_usage'] = $scope.selectedUsages;
+            }
```

Comment:
Could DRY this up a bit by having a function like
```js
$scope.selectedItemToParam = function (item) {
  var mapping = { 
     "None (Clear)": "",
     "No change": null,
  };
  if (mapping.hasOwnProperty(item) {
    return mapping[item];
  }
  return item;
}
```
Then you could just say 
```js
params['purpose_usage'] = $scope.selectedItemToParam($scope.selectedUsages);
params['purpose_role'] = $scope.selectedItemToParam($scope.selectedRoles);
```
etc.